src/Common/src/System/IO/RowConfigReader.cs

```diff
@@ -82,7 +82,11 @@ public bool TryGetNextValue(string key, out string value)
             // after. This is the format of most "row-based" config files in /proc/net, etc.
             int afterKey = keyIndex + key.Length;
             int endOfLine = _buffer.IndexOf(Environment.NewLine, afterKey, _comparisonKind);
-            Debug.Assert(endOfLine != -1, "RowConfigReader needs a newline after the key, and one was not found.");
+            if (endOfLine == -1)
+            {
+                // There may not be a newline after this key, if we've reached the end of the file.
+                endOfLine = _buffer.Length - 1;
+            }
```

Comment:
What would happen if you had the string " key"... would it end up saying that the value was the same as the key, because it would search back and find the space before the key?